app/src/main/java/org/mozilla/reference/browser/browser/FirefoxAccountsIntegration.kt

```diff
@@ -4,115 +4,123 @@
 
 package org.mozilla.reference.browser.browser
 
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.OnLifecycleEvent
 import android.content.Context
 import android.content.SharedPreferences
 import android.net.Uri
-import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.CoroutineScope
+import kotlinx.coroutines.experimental.Deferred
+import kotlinx.coroutines.experimental.Dispatchers
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.async
 import kotlinx.coroutines.experimental.launch
+import kotlinx.coroutines.experimental.runBlocking
 import mozilla.components.concept.engine.EngineSession
 import mozilla.components.concept.engine.request.RequestInterceptor
 import mozilla.components.feature.tabs.TabsUseCases
 import mozilla.components.service.fxa.Config
 import mozilla.components.service.fxa.FirefoxAccount
-import mozilla.components.service.fxa.OAuthInfo
+import mozilla.components.service.fxa.FxaException
 import mozilla.components.service.fxa.Profile
+import kotlin.coroutines.experimental.CoroutineContext
 
 class FirefoxAccountsIntegration(
     private val context: Context,
     private val tabsUseCases: TabsUseCases
-) {
+) : CoroutineScope, LifecycleObserver {
+
     companion object {
-        const val CLIENT_ID = "12cc4070a481bc73"
-        const val REDIRECT_URL = "fxaclient://android.redirect"
-
-        // TODO get URL from Config object: https://github.com/mozilla/application-services/issues/305
-        const val CONFIG_URL = "https://latest.dev.lcip.org"
-
-        // TODO redirect to URL directly: https://github.com/mozilla-mobile/android-components/issues/1106
-        const val SUCCESS_PAGE = """
-            <head>
-                <meta
-                    http-equiv="refresh"
-                    content="0; URL=$CONFIG_URL/connect_another_device?showSuccessMessage=true" />
-            </head>
-        """
+        const val CLIENT_ID = "3c49430b43dfba77"
+        const val REDIRECT_URL = "https://accounts.firefox.com/oauth/success/3c49430b43dfba77"
+        const val SUCCESS_PATH = "connect_another_device?showSuccessMessage=true"
```

Comment:
Eventually, we'll aim to have the above `REDIRECT_URL` show the success message directly so that you don't need a separate `SUCCESS_PATH` page here.

Reply:
👍